### ISSUE TITLE
Problems with the directory dependency graph

### ISSUE THREAD

**Issue by albert-github** [doxygen/doxygen#9386]
With the documentation of the blender package as generated by Fossies (https://fossies.org/dox/blender-3.2.0/) we get the messages:
```
Warning: flat edge between adjacent nodes one of which has a record shape - replace records with HTML-like labels
  Edge dir_999b5e1ba5bd3f79e8f5909eb5c2b706 -> dir_990b19eeab27163043c97562d7574900
Error: getsplinepoints: no spline points available for edge (dir_999b5e1ba5bd3f79e8f5909eb5c2b706,dir_990b19eeab27163043c97562d7574900)
Error: lost dir_999b5e1ba5bd3f79e8f5909eb5c2b706 dir_990b19eeab27163043c97562d7574900 edge
```
Note that Fossies is using the current `dot` version 4.0.0

When we look into the problem we see that we have here in the beginning:
```
node [fontname="Helvetica",fontsize="10",shape=record];
```
and further on lines like:
```
dir_201221c68d7bad0112972703f119f842 [label="blendthumb", style="filled,", fillcolor="#a2b4d6", color="grey25", URL="dir_201221c68d7bad0112972703f119f842.html"];
```
in the past these lines looked like:
```
dir_201221c68d7bad0112972703f119f842 [shape="box",label="blendthumb", style="filled,", fillcolor="#a2b4d6", color="grey25", URL="dir_201221c68d7bad0112972703f119f842.html"];
```
(so with an explicit `shape="box"`.

When I change by hand the `shape` to `box` the warning disappears.

**Files**
The dot files and resulting svg files (dot 4.0.0 and 2.38) with `record` and with `box`: [dot_dir_graph.tar.gz](https://github.com/doxygen/doxygen/files/8869767/dot_dir_graph.tar.gz)

@makelinux  do you think:
- is this a problem in the definition of the `dot` language 

or
- a general problem in doxygen 

or
- is this a bug in `dot` version 4.0.0

**Comment by albert-github** [comment]
Some further test revealed that the problem already surfaced in the dot version 2.40

So it is probably a general dot situation and the `shape` `record` should not be used for these used for these situations.

**Comment by makelinux** [comment]
I suppose it is dot memory access bug. There are some issues.
It works without line 

  dir_999b5e1ba5bd3f79e8f5909eb5c2b706->dir_990b19eeab27163043c97562d7574900 [headlabel="6", labeldistance=1.5 headhref="dir_000053_000200.html"];

**Comment by albert-github** [comment]
@makelinux what do you propose to do with it (seen that `box` is working  and the fact that the the documentation of `dot` (https://graphviz.org/doc/info/shapes.html) says:
> The record-based shape has largely been superseded and greatly generalized by [HTML-like labels](#html). That is, instead of using `shape=record`, one might consider using `shape=none`, `margin=0` and an HTML-like label. 

Looks a bit like that `shape=record` is more intended for things like:
![image](https://user-images.githubusercontent.com/5223533/172887159-bc8d7ab0-a09f-4daf-ba6b-cfc2be2a30a5.png)


and

![image](https://user-images.githubusercontent.com/5223533/172887217-f4c571c2-bea3-4014-ba74-7000d52c7253.png)


and the `shape=box` or shape=rectangle` (`shape=rect`) are more for the doxygen usage.
One type that might be a bit problematic is when we have a box in a box like:

![image](https://user-images.githubusercontent.com/5223533/172888561-a4c8d5b3-8c75-4e32-893f-50555cdd9008.png)

**Comment by makelinux** [comment]
I've opened https://gitlab.com/graphviz/graphviz/-/issues/2248

**Comment by makelinux** [comment]
> and the `shape=box` or shape=rectangle` (`shape=rect`) are more for the doxygen usage.

I likely agree.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9390.

**Comment by makelinux** [comment]
Erroneous edge:
![Screenshot from 2022-06-10 20-08-35](https://user-images.githubusercontent.com/2335185/173117138-d5c0ae2f-65a6-46dc-a1a1-9a3ead65164d.png)

**Comment by albert-github** [comment]
I got an email saying you found an "Erroneous edge:" but I don't see anything here, did you delete the comment?

**Comment by makelinux** [comment]
@albert-github , fixed.

Erroneous horizontal **flat edge**  between nodes of same rank:: 
![Screenshot from 2022-06-10 20-33-49](https://user-images.githubusercontent.com/2335185/173120586-afc5523e-d3ad-4ede-840e-d9be352f1f9d.png)

**Comment by albert-github** [comment]
Do you mean that there is an extra link in image?
I assume that this line is present in the original dot file.
- is it only present when using `shape="box"` or also when `shape="record"`?
- when present in both cases it has nothing to do with this issue and a new issue is required with which the problem can be reproduced

**Comment by makelinux** [comment]
It present only using `shape="box"`. I just colored it red. 
With `shape="record"` the edge is removed. See error message `Error: lost ... edge`

**Comment by albert-github** [comment]
Clear, in your opinion should the edge be present or not?

**Comment by makelinux** [comment]
It should be present.

**Comment by albert-github** [comment]
So actually the 'shape="box"` is correct, which would be logical due the error message and an argument more to use `shape="box"`.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.5.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
